src/contracts/core/AllocationManager.sol

```diff
@@ -0,0 +1,731 @@
+     * @param allocation the magnitude allocations to modify for a single strategy
+     * @param allocationEffectTimestamp the timestamp when the allocations will take effect
+     * @param deallocationCompletableTimestamp the timestamp when the deallocations will be completable
+     * @dev For each allocation, allocation.operatorSets MUST be ordered in ascending order according to the
+     * encoding of the operatorSet. This is to prevent duplicate operatorSets being passed in. The easiest way to ensure
+     * ordering is to sort allocated operatorSets by address first, and then sort for each avs by ascending operatorSetIds.
+     */
+    function _modifyAllocations(
+        address operator,
+        MagnitudeAllocation calldata allocation,
+        uint32 allocationEffectTimestamp,
+        uint32 deallocationCompletableTimestamp
+    ) internal {
+        require(allocation.operatorSets.length == allocation.magnitudes.length, InputArrayLengthMismatch());
+
+        for (uint256 i = 0; i < allocation.operatorSets.length; ++i) {
+            require(
+                avsDirectory.isOperatorSet(allocation.operatorSets[i].avs, allocation.operatorSets[i].operatorSetId),
+                InvalidOperatorSet()
+            );
+
+            bytes32 operatorSetKey = _encodeOperatorSet(allocation.operatorSets[i]);
+            // prep magnitudeUpdates in storage
+
+            // Check that there is at MOST `MAX_PENDING_UPDATES` combined allocations & deallocations for the operator, operatorSet, strategy
+            (uint256 numPendingAllocations, uint256 numPendingDeallocations) =
+                _getPendingAllocationsAndDeallocations(operator, allocation.strategy, operatorSetKey);
+
+            require(
+                numPendingAllocations + numPendingDeallocations < MAX_PENDING_UPDATES,
+                PendingAllocationOrDeallocation()
+            );
+
+            Snapshots.History storage magnitudeUpdates = _magnitudeUpdate[operator][allocation.strategy][operatorSetKey];
+
+            // Read current magnitude allocation including its respective array index and length.
+            // We'll use these values later to check the number of pending allocations/deallocations.
+            (uint224 currentMagnitude,,) =
+                magnitudeUpdates.upperLookupRecentWithPos(uint32(block.timestamp));
+            require(currentMagnitude != allocation.magnitudes[i], SameMagnitude());
+
+            if (allocation.magnitudes[i] < uint64(currentMagnitude)) {
+                // Newly configured magnitude is less than current value.
+                // Therefore we handle this as a deallocation
+
+                // Note: MAX_PENDING_UPDATES == 1, so we do not have to decrement any allocations
+
+                // 1. push PendingFreeMagnitude and respective array index into (op,opSet,Strategy) queued deallocations
+                uint64 magnitudeToDeallocate = uint64(currentMagnitude) - allocation.magnitudes[i];
```

Comment:
General complaint with this contract: "magnitudes" are expressed using either `uint224` OR `uint64`, depending on whatever the author decided when they wrote the line of code.

We should consistently use a single type to hold magnitudes.

Reply:
While we typically use uint64 for magnitude values, lib `Snapshots` uses uint224 for snapshot values.

Let's discuss further before I make changes.